src/crypto/signature.js

```diff
@@ -117,13 +117,21 @@ export async function verify(algo, hashAlgo, signature, publicParams, data, hash
  * @param {Object} privateKeyParams - Algorithm-specific public and private key parameters
  * @param {Uint8Array} data - Data to be signed
  * @param {Uint8Array} hashed - The hashed data
+ * @param {Object} [config] - Custom configuration settings to overwrite those in [config]{@link module:config}
  * @returns {Promise<Object>} Signature                      Object containing named signature parameters.
  * @async
  */
-export async function sign(algo, hashAlgo, publicKeyParams, privateKeyParams, data, hashed) {
+export async function sign(algo, hashAlgo, publicKeyParams, privateKeyParams, data, hashed, config = null) {
```

Comment:
```suggestion
export async function sign(algo, hashAlgo, publicKeyParams, privateKeyParams, data, hashed, config) {
```
Is there a reason to have it `null` ? I saw the same in other places. 
The config should never be undefined, if needed, it's assumed to be defined, possibly set to `defaultConfig` by an higher level function.
This also means you can then remove the `config &&`  checks everywhere 🙂

Reply:
The idea was to keep the API backwards compatibility, and I initially missed the `defaultConfig` way of specifying the value. That clears implementation a bit, good spot!

Reply:
This should be corrected now - resolving.